### DISCUSSION TITLE
Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'BoringSSL-GRPC' from project 'Pods')

### DISCUSSION THREAD

**Discussion by saviobraz** [codemagic-ci-cd/codemagic-docs#2862] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Dando erro ao compilar

### Expected results

não sei o motivo do erro, estou há dias pesquisando, poderiam me ajudar

### Actual results

Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'BoringSSL-GRPC' from project 'Pods')

### Build id (optional)

674dd7db71fd219e4b576a62

**Comment by dam-ease** [comment]
Hello @saviobraz,
This is the error output from your build logs
```
Error output from Xcode build:
↳
    ** ARCHIVE FAILED **


Xcode's output:
↳
    Writing result bundle at path:
    	/var/folders/w2/rrf5p87d1bbfyphxc7jdnyvh0000gn/T/flutter_tools.p4RSxy/flutter_ios_build_temp_dir9p1SZP/temporary_xcresult_bundle

    /Users/builder/.pub-cache/hosted/pub.dev/url_launcher_ios-6.3.0/ios/url_launcher_ios/Sources/url_launcher_ios/Launcher.swift:22:1: error: type 'UIApplication' does not conform to protocol 'Launcher'
    extension UIApplication: Launcher {}
    ^
    UIKit.UIApplication:24:15: note: candidate has non-matching type '(URL, [UIApplication.OpenExternalURLOptionsKey : Any], (@MainActor @Sendable (Bool) -> Void)?) -> Void'
        open func open(_ url: URL, options: [UIApplication.OpenExternalURLOptionsKey : Any] = [:], completionHandler completion: (@MainActor @Sendable (Bool) -> Void)? = nil)
                  ^
    /Users/builder/.pub-cache/hosted/pub.dev/url_launcher_ios-6.3.0/ios/url_launcher_ios/Sources/url_launcher_ios/Launcher.swift:15:8: note: protocol requires function 'open(_:options:completionHandler:)' with type '(URL, [UIApplication.OpenExternalURLOptionsKey : Any], ((Bool) -> Void)?) -> ()'; add a stub for conformance
      func open(
           ^
    Command SwiftCompile failed with a nonzero exit code
``` 
The issue is related to the `url_launcher_ios` package, you can check out this [issue](https://github.com/flutter/flutter/issues/151467) and the fix is currently in the latest version of the package, kindly upgrade to the latest version of the `url_launcher_ios` package [6.3.1](https://pub.dev/packages/url_launcher_ios/changelog#631)

**Reply by saviobraz** [reply]
Atualizei o pubspec.yaml para
    cloud_functions: ^5.1.5
    firebase_core: ^3.8.0 
    firebase_auth: ^5.3.3
    firebase_storage: ^12.3.6
    cloud_firestore: ^5.5.0
 url_launcher: ^6.3.1 e removi pubspec.lock, mas continua o erro com ID da compilação:674f161513f9df4e58fc7ae7

 note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'GoogleDataTransport' from project 'Pods')
    note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'Google-Maps-iOS-Utils' from project 'Pods')
    note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'GTMSessionFetcher-GTMSessionFetcher_Core_Privacy' from project 'Pods')
    note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'GTMSessionFetcher' from project 'Pods')
    note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'FirebaseStorage' from project 'Pods')
    note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'FirebaseSharedSwift' from project 'Pods')
    note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'FirebaseSessions' from project 'Pods')
    note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'FirebaseRemoteConfigInterop' from project 'Pods')
    note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'FirebaseRemoteConfig-FirebaseRemoteConfig_Privacy' from project 'Pods')
    note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'FirebaseRemoteConfig' from project 'Pods')
    note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'FirebaseMessagingInterop' from project 'Pods')
    note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'FirebaseMessaging-FirebaseMessaging_Privacy' from project 'Pods')
    note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'FirebaseMessaging' from project 'Pods')
    note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'FirebaseInstallations-FirebaseInstallations_Privacy' from project 'Pods')
    note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'FirebaseInstallations' from project 'Pods')
    note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'FirebaseFunctions' from project 'Pods')
    note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'FirebaseFirestoreInternal-FirebaseFirestoreInternal_Privacy' from project 'Pods')
    note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'FirebaseFirestoreInternal' from project 'Pods')
    note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'FirebaseFirestore-FirebaseFirestore_Privacy' from project 'Pods')
    note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'FirebaseFirestore' from project 'Pods')
    note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'FirebaseCrashlytics-FirebaseCrashlytics_Privacy' from project 'Pods')
    note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'FirebaseCrashlytics' from project 'Pods')
    note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'FirebaseCoreInternal-FirebaseCoreInternal_Privacy' from project 'Pods')
    note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'FirebaseCoreInternal' from project 'Pods')
    note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'FirebaseCoreExtension-FirebaseCoreExtension_Privacy' from project 'Pods')
    note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'FirebaseCoreExtension' from project 'Pods')
    note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'FirebaseCore-FirebaseCore_Privacy' from project 'Pods')
    note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'FirebaseCore' from project 'Pods')
    note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'FirebaseAuthInterop' from project 'Pods')
    note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'FirebaseAuth-FirebaseAuth_Privacy' from project 'Pods')
    note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'FirebaseAuth' from project 'Pods')
    note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'FirebaseAppCheckInterop' from project 'Pods')
    note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'FirebaseAppCheck' from project 'Pods')
    note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'FirebaseABTesting-FirebaseABTesting_Privacy' from project 'Pods')
    note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'FirebaseABTesting' from project 'Pods')
    /Users/builder/clone/ios/Pods/Pods.xcodeproj: warning: The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 10.0, but the range of supported deployment target versions is 12.0 to 18.2.99. (in target 'BoringSSL-GRPC-openssl_grpc' from project 'Pods')
    note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'BoringSSL-GRPC-openssl_grpc' from project 'Pods')
    /Users/builder/clone/ios/Pods/Pods.xcodeproj: warning: The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 10.0, but the range of supported deployment target versions is 12.0 to 18.2.99. (in target 'BoringSSL-GRPC' from project 'Pods')
    note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'BoringSSL-GRPC' from project 'Pods')
    warning: Run script build phase 'Create Symlinks to Header Folders' will be run during every build because it does not specify any outputs. To address this issue, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'BoringSSL-GRPC' from project 'Pods')
    note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'AppCheckCore' from project 'Pods')
    note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'Firebase' from project 'Pods')
    note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'Flutter' from project 'Pods')

Encountered error while archiving for device.


Build failed :|
Failed to build for iOS

Publishing

**Reply by wade-beavers** [reply]
Same issue for 674f7da94ede54c8c3d3c9d0

**Reply by dam-ease** [reply]
Hello @saviobraz.
In both logs, there is the error message below:
```
warning: 'keyWindow' was deprecated in iOS 13.0: Should not be used for applications that support multiple scenes as it returns a key window across all connected scenes
                let controller: UIViewController? = UIApplication.shared.keyWindow?.rootViewController
```
I did a quick search and found an existing [thread](https://stackoverflow.com/questions/57134259/how-to-resolve-keywindow-was-deprecated-in-ios-13-0) on Stack Overflow. Please can you try out the suggested solutions in this thread, I think it should solve the issue in this case as well.

**Reply by saviobraz** [reply]
lembrando que estou usando flutter usando codemagic, pois já fiz vários uploads para apple store e sempre funcionou, não motivos do erros no momento, continuo pesquisando a solução do problema, mas continua o erro.

**Reply by dam-ease** [reply]
@saviobraz In your recent build with buildID - 6750511665e91e06e73d8200, the error below seems to be causing the failure:
```
 error: ../.pub-cache/hosted/pub.dev/esc_pos_printer_plus-0.0.3/lib/src/network_printer.dart:162:26: Error: Member not found: 'Size4'.
        QRSize size = QRSize.Size4
 ```
which has been reported in the esc_pos_printer repository. The issue for reference - https://github.com/andrey-ushakov/esc_pos_printer/issues/200. Please can you also try out the suggested solution [here](https://github.com/andrey-ushakov/esc_pos_printer/issues/200#issuecomment-2503581674)

**Reply by dam-ease** [reply]
Hello @wade-beavers ,
In your logs, I see an error related to flutterfire which is different from that of the original issue.
```
/Users/builder/Library/Developer/Xcode/DerivedData/Runner-edaimyiflreloheqntgnhkmwcclv/Build/Intermediates.noindex/ArchiveIntermediates/staging/IntermediateBuildFilesPath/Runner.build/Release-staging-iphoneos/Runner.build/Script-C0C64AAD78E6962510A2AF43.sh: line 5: flutterfire: command not found
    Command PhaseScriptExecution failed with a nonzero exit code
```
This issue seems to be an already reported issue with flutterfire_cli dependency. The reported issue in flutterfire_cli tool repo for reference: https://github.com/invertase/flutterfire_cli/issues/257. Can you try to add "dart pub global activate flutterfire_cli" to the prebuild script section in Codemagic?

**Reply by dam-ease** [reply]
Hello @saviobraz @wade-beavers,
I would like to check-in on how this is going on your end?
Have you been able to try the suggested solutions out?
Please if you have any more blockers or issues, kindly let me know. Glad to help!

**Reply by saviobraz** [reply]
Alterei a versão do XCode de 16 para 15 e parou de dar o erro mencionado, mas agora mostra o erro Failed to build iOS app
Error (Xcode): Undefined symbols:

Error (Xcode): Linker command failed with exit code 1 (use -v to see invocation)

Encountered error while archiving for device.


Build failed :|
ID da Compilação: 6751eb7ed381b2ef7927811d

**Reply by dam-ease** [reply]
Thanks for your response @saviobraz.
I checked through your logs and see you are making use of Xcode version 15.0.1 with Firebase. Firebase requires at least Xcode 15.2. See the [release notes](https://firebase.google.com/support/release-notes/ios). Please can you make use of an higher version and share how it goes on your end?

**Comment by saviobraz** [comment]
Resolvido, era a versão do XCode, mas com XCode 16 ocorre o erro. Funcionou
com a versão 15.1. Obrigado pela ajuda.

Em ter., 17 de dez. de 2024 06:01, Damilola Alimi ***@***.***>
escreveu:
